app/controllers/api/v0/swagger/models/highlight.rb

```diff
@@ -114,6 +116,19 @@ module Api::V0::Swagger::Models::Highlight
     end
   end
 
+  add_properties(:HighlightUpdate) do
+    property :color do
+      key :type, :string
+      # remove the anchors because swagger-codegen always escapes them
+      key :pattern, ::Highlight::VALID_COLOR.inspect[1..-2]
+      key :description, 'The highlight color.'
+    end
+    property :annotation do
```

Comment:
can you send in an update with neither of these properties? (e.g., one is required?) . or are both required?

Reply:
Yes you can, in which case nothing will change.